File: server/myClient/Assets/myScript/entity/Point.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Assets.myScript.entity
{
    public class Point
    {
        public string name { get; set; }
        public string type { get; set; }
        public float x { get; set; }
        public float y { get; set; }
        public float size_w { get; set; }
        public float size_h { get; set; }
        public string description { get; set; }
        public int id_map { get; set; }
        public int busy { get; set; }
        public int id_user_Busy { get; set; }
        public int id { get; set; }
        public int all_space { get; set; }
        public int free_space { get; set; }

        public Point(string name, string type, float x, float y, float size_w, float size_h, string description, int id_map, int busy, int id_user_Busy, int id)
        {
            this.name = name;
            this.type = type;
            this.description = description;
            this.x = x;
            this.id_map = id_map;
            this.id = id;
            this.y = y;
            this.busy = busy;
            this.id_user_Busy = id_user_Busy;
            this.size_w = size_w;
            this.size_h = size_h;
            all_space = 0;
        }

        public Point(string name, string type, string description, int id, int id_user_Busy, int all_space)
        {
            this.all_space = all_space;
            this.name = name;
            this.type = type;
            this.description = description;
            this.id = id;
            this.id_user_Busy = id_user_Busy;
            this.x = 0;
            this.id_map = 0;
            this.y = 0;
            this.busy = 0;
            this.size_w = 0;
            this.size_h = 0;

        }

        public Point(int id, int id_user_Busy, int busy, string name, string type, string description)
        {
            this.name = name;
            this.type = type;
            this.description = description;
            this.id = id;
            this.id_user_Busy = id_user_Busy;
            this.x = 0;
            this.id_map = 0;
            this.y = 0;
            this.busy = busy;
            this.size_w = 0;
            this.size_h = 0;
            all_space = 0;

        }

        public Point()
        {
            this.name = "";
            this.type = "ACTIV";
            this.description = "";
            this.x = 0;
            this.id_map = 0;
            this.id = 0;
            this.y = 0;
            this.busy = 0;
            this.id_user_Busy = 0;
            this.size_w = 0;
            this.size_h = 0;
            all_space = 0;

        }

        public Point(string type)
        {
            this.name = "";
            this.type = type;
            this.description = "";
            this.x = 0;
            this.id_map = 0;
            this.id = 0;
            this.y = 0;
            this.busy = 0;
            this.id_user_Busy = 0;
            this.size_w = 0;
            this.size_h = 0;
            all_space = 0;

        }

        public bool Equals(Object obj) 
        {
            if (this == obj)
                return true;
            if (obj == null)
                return false;
            Point other = (Point)obj;
            if (id != other.id)
                return false;

            if (busy != other.busy)
                return false;
            if (id_user_Busy != other.id_user_Busy)
                return false;

            if (y != other.y)
                return false;
            if (x != other.x)
                return false;

            if (size_w != other.size_w)
                return false;
            if (size_h != other.size_h)
                return false;
            return true;
        }

    }
}

File: отчёты/код/server/Point.java
package com.SERV.view.entity;

import com.fasterxml.jackson.annotation.JsonAutoDetect;

import java.io.Serializable;

/**
 * Created by Prizrak on 08.07.2014.
 */
@JsonAutoDetect
public class Point implements Serializable{
    private String name;
    private String type;
    private double x;
    private double y;
    private double size_w;
    private double size_h;
    private String description;
    private int id_map;
    private int busy;
    private int id_user_Busy;
    private int id;
    private int all_space;
    private int free_space;

    public Point(String name, String type, double x, double y, double size_w, double size_h, String description, int id_map, int busy, int id_user_Busy, int all_space, int id, int free_space) {
        this.name = name;
        this.type = type;
        this.x = x;
        this.y = y;
        this.size_w = size_w;
        this.size_h = size_h;
        this.description = description;
        this.id_map = id_map;
        this.busy = busy;
        this.id_user_Busy = id_user_Busy;
        this.all_space = all_space;
        this.id = id;
        this.free_space = free_space;
    }

    public Point(String name, String type, double x, double y, double size_w, double size_h, String description, int id_map, int busy, int id_user_Busy, int all_space, int id) {
        this.name = name;
        this.type = type;
        this.x = x;
        this.y = y;
        this.size_w = size_w;
        this.size_h = size_h;
        this.description = description;
        this.id_map = id_map;
        this.busy = busy;
        this.id_user_Busy = id_user_Busy;
        this.all_space = all_space;
        this.id = id;
        this.free_space = 0;
    }

    public Point(){}

    public int getAll_space() {
        return all_space;
    }

    public void setAll_space(int all_space) {
        this.all_space = all_space;
    }

    public int getFree_space() {
        return free_space;
    }

    public void setFree_space(int free_space) {
        this.free_space = free_space;
    }

    public String getName() {
        return name;
    }
    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }
    public void setDescription(String description) {
        this.description = description;
    }

    public String getType() {
        return type;
    }
    public void setType(String type) {
        this.type = type;
    }

    public double getX() {
        return x;
    }
    public void setX(double x) {
        this.x = x;
    }

    public int getId_map() {
        return id_map;
    }
    public void setId_map(int id_map) {
        this.id_map = id_map;
    }

    public double getY() {
        return y;
    }
    public void setY(double y) {
        this.y = y;
    }

    public double getSize_w() {
        return size_w;
    }
    public void setSize_w(double size_w) {
        this.size_w = size_w;
    }

    public double getSize_h() {
        return size_h;
    }
    public void setSize_h(double size_h) {
        this.size_h = size_h;
    }

    public int getBusy() {return busy;}
    public void setIsBusy(int busy) {
        this.busy=busy;
    }

    public void setBusy(int busy) {
        this.busy = busy;
    }

    public int getId_user_Busy() {
        return id_user_Busy;
    }
    public void setId_user_Busy(int id_user_Busy) {
        this.id_user_Busy = id_user_Busy;
    }



    @Override
    public int hashCode(){
        return id;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj)
            return true;
        if (obj == null)
            return false;
        if (getClass() != obj.getClass())
            return false;
        Point other = (Point) obj;
        if (id!=((Point) obj).id)
            return false;
        return true;
    }

    public void setId(int id) {
        this.id = id;
    }
    public int getId() {
        return id;
    }
}
